### ISSUE TITLE
Parea wrapper not re-raising root exception

### ISSUE THREAD

**Issue by seanr-cardless** [parea-ai/parea-sdk-py#166]
## 🐛 Bug Report

The Parea wrapper code returns in the `finally` block which swallows the exception caught when actually calling the OpenAI models. This makes things very difficult to log/monitor/debug as the root exception is swallowed by Parea and failures occur downstream.

Link to offending code.

https://github.com/parea-ai/parea-sdk-py/blob/bffcada8fe534fd7d2ea543a61322749d20ef4ca/parea/wrapper/wrapper.py#L137

## 🔬 How To Reproduce

1. I'm not sure how to repro an OpenAI failure but it can be monkey patched if needed. Looking at the offending code probably provides all the context necessary 

### Code sample

Try running this function. 1 is returned and the exception isn't raised

```python
def run():
    try:
        raise Exception("bad")
    except Exception as e:
        print(e)
        raise e
    finally:
        return 1


```

### Environment

* OS: MacOS apple silicon 
* Python version: 3.9

```bash
python --version
```

## 📈 Expected behavior

The error isn't swallowed by Parea and is surfaced to the consumer of the Open API call.

## 📎 Additional context

I ran into this using Langchain with the following (abbreviated) code

```python3
    llm = ChatOpenAI(
        openai_api_key=openai_api_key,
        temperature=0,
        model_name=model,
        model_kwargs=llm_kwargs,
        max_retries=3
    )
    response = llm([HumanMessage(content="model query here...any will work")
```

This became relevant on the OpenAI [outage](https://status.openai.com/) on 2023-10-19

**Comment by joschkabraun** [comment]
Hey, thanks for raising! I will update you once it's resolved.

**Comment by joschkabraun** [comment]
@seanr-cardless it's fixed now with #168!
